src/main/java/bio/terra/service/filedata/flight/ingest/IngestCopyLoadHistoryToBQStep.java

```diff
@@ -0,0 +1,95 @@
+    private final DatasetService datasetService;
+    private final String loadTag;
+    private final String datasetIdString;
+    private final BigQueryPdao bigQueryPdao;
+    private final int fileChunkSize;
+
+    public IngestCopyLoadHistoryToBQStep(LoadService loadService,
+                                         DatasetService datasetService,
+                                         String loadTag,
+                                         String datasetId,
+                                         BigQueryPdao bigQueryPdao,
+                                         int fileChunkSize) {
+        this.loadService = loadService;
+        this.loadTag = loadTag;
+        this.datasetIdString = datasetId;
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+        this.fileChunkSize = fileChunkSize;
+    }
+
+    @Override
+    public StepResult doStep(FlightContext context) {
+        FlightMap workingMap = context.getWorkingMap();
+        String loadIdString = workingMap.get(LoadMapKeys.LOAD_ID, String.class);
+        UUID loadId = UUID.fromString(loadIdString);
+        UUID datasetId = UUID.fromString(datasetIdString);
+        Dataset dataset = datasetService.retrieve(datasetId);
+
+        int chunkNum = 0;
+        List<BulkLoadHistoryModel> loadHistoryArray = null;
+        String flightId = context.getFlightId();
+        String tableName_FlightId = flightId.replaceAll("[^a-zA-Z0-9]", "_");
+        try {
+            Instant loadTime = context.getStairway().getFlightState(flightId).getSubmitted();
+            bigQueryPdao.createStagingLoadHistoryTable(dataset, tableName_FlightId);
+
+            while (loadHistoryArray == null || loadHistoryArray.size() == fileChunkSize) {
+                loadHistoryArray = loadService.makeLoadHistoryArray(loadId, fileChunkSize, (chunkNum * fileChunkSize));
+                chunkNum++;
+                // send list plus load_tag, load_time to BQ to be put in a staging table
+                if (!loadHistoryArray.isEmpty()) {
+                    bigQueryPdao.loadHistoryToStagingTable(
+                        dataset,
+                        tableName_FlightId,
+                        loadTag,
+                        loadTime,
+                        loadHistoryArray);
+                }
+            }
+            // copy from staging to actual BQ table
+            bigQueryPdao.mergeStagingLoadHistoryTable(dataset, tableName_FlightId);
+            bigQueryPdao.deleteStagingLoadHistoryTable(dataset, tableName_FlightId);
+        } catch (Exception ex) {
+            logger.error("Failure deleting load history staging table for flight: " + flightId, ex);
+        }
+
+        return StepResult.getStepResultSuccess();
+    }
+
+    @Override
+    public StepResult undoStep(FlightContext context) {
+        String flightId = context.getFlightId();
+        try {
+            // not sure if I should move this stuff into a helper method?
```

Comment:
I think this is only being used once it isn't necessary to move it into a helper method.

Reply:
Thank you for the reminder to remove this :)